### ISSUE TITLE
about the network

### ISSUE THREAD

**Issue by skygoo** [soubhiksanyal/RingNet#23]
hello ,Great work! Will you open source the train code ?

**Comment by skygoo** [comment]
I have a question during reading the paper. Did the R rings elements mean there is one network(encoder), and for each step, use the same network(encoder) compute(encode) R images, the use the R results compute Lsc.

**Comment by soubhiksanyal** [comment]
1) The trick is in batching the data. yes the concept is similar like Siamese networks extended to rings. One can pass all the data through one encoder and slice the data according to their labels at the end of the encoder output to consider it for different rings and compute the loss.
2) Only the 100 shape related vectors are considered.

**Comment by soubhiksanyal** [comment]
Releasing the training code is having some internal licensing issues due to its reliance in tensorflow flame which may take further time.
